File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/support/Failure.java
package fr.myprysm.vertx.elasticsearch.action.support;

import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@DataObject(generateConverter = true)
public class Failure {
    private String cause;

    public Failure(JsonObject json) {
        FailureConverter.fromJson(json, this);
    }

    public JsonObject toJson() {
        JsonObject json = new JsonObject();
        FailureConverter.toJson(this, json);
        return json;
    }

}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/support/Failure.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.support

import fr.myprysm.vertx.elasticsearch.action.support.Failure

fun Failure(
  cause: String? = null): Failure = fr.myprysm.vertx.elasticsearch.action.support.Failure().apply {

  if (cause != null) {
    this.setCause(cause)
  }
}

